### PR TITLE
AUX2 Serial driver with idle callback

### PR DESCRIPTION
**Pull request by ajjjjjjjj**
- [x] TX test
- [x] RX test
- [x] IdleCb test

### CODE REVIEW

File: radio/src/targets/common/arm/stm32/aux_serial_driver.cpp [hunk 1]
```diff
@@ -256,3 +249,124 @@ extern "C" void AUX_SERIAL_USART_IRQHandler(void)
...
+  USART_InitStructure.USART_StopBits = stop;
+  USART_InitStructure.USART_Parity = parity;
+  USART_InitStructure.USART_HardwareFlowControl = USART_HardwareFlowControl_None;
+  USART_InitStructure.USART_Mode = USART_Mode_Tx | USART_Mode_Rx;
+  USART_Init(AUX2_SERIAL_USART, &USART_InitStructure);
+
+  // if (dma) {
+    aux2SerialRxFifo.stream = AUX2_SERIAL_DMA_Channel_RX; // workaround, CNDTR reading do not work otherwise
+    DMA_InitTypeDef DMA_InitStructure;
+    auxSerialRxFifo.clear();
```
Comment (rafolg) [inline radio/src/targets/common/arm/stm32/aux_serial_driver.cpp:287]: This line should be aux2SerialRxFifo.clear(); not auxSerialRxFifo.clear(); ---- the number "2" is missing

File: radio/src/targets/flysky/hal.h [hunk 2]
```diff
@@ -451,6 +451,21 @@ extern void ISR_TIMER3_CAPT_vect(void);
...
+#define AUX2_SERIAL_RCC_AHB1Periph         (RCC_AHBPeriph_GPIOC | RCC_AHBPeriph_DMA1)
+#define AUX2_SERIAL_RCC_APB1Periph         RCC_APB1Periph_USART4
+#define AUX2_SERIAL_GPIO                   GPIOC
+#define AUX2_SERIAL_GPIO_PIN_TX            GPIO_Pin_10 // PC10
+#define AUX2_SERIAL_GPIO_PIN_RX            GPIO_Pin_11 // PC11
+#define AUX2_SERIAL_GPIO_PinSource_TX      GPIO_PinSource10
+#define AUX2_SERIAL_GPIO_PinSource_RX      GPIO_PinSource11
+#define AUX2_SERIAL_GPIO_AF                GPIO_AF_0
+#define AUX2_SERIAL_USART                  USART4
+#define AUX2_SERIAL_USART_IRQHandler       USART4_IRQHandler
```
Comment (rafolg) [inline radio/src/targets/flysky/hal.h:464]: USART4_IRQHandler does not exist for F0, it should be USART3_4_IRQHandler

File: radio/src/targets/flysky/hal.h [hunk 3]
```diff
@@ -451,6 +451,21 @@ extern void ISR_TIMER3_CAPT_vect(void);
...
+#define AUX2_SERIAL_USART_IRQn             USART4_IRQn
```
Comment (rafolg) [inline radio/src/targets/flysky/hal.h:465]: Same for this line, USART4_IRQn does not exist for F0, it should be USART3_4_IRQn

File: radio/src/targets/common/arm/stm32/aux_serial_driver.cpp [hunk 4]
```diff
@@ -256,3 +249,124 @@ extern "C" void AUX_SERIAL_USART_IRQHandler(void)
...
+    USART_ITConfig(AUX2_SERIAL_USART, USART_IT_RXNE, DISABLE);
+    USART_ITConfig(AUX2_SERIAL_USART, USART_IT_TXE, DISABLE);
+
+    DMA_InitStructure.DMA_PeripheralBaseAddr = CONVERT_PTR_UINT(&AUX2_SERIAL_USART->RDR);
+    DMA_InitStructure.DMA_MemoryBaseAddr = CONVERT_PTR_UINT(aux2SerialRxFifo.buffer());
+    DMA_InitStructure.DMA_DIR = DMA_DIR_PeripheralSRC;
+    DMA_InitStructure.DMA_BufferSize = auxSerialRxFifo.size();
```
Comment (rafolg) [inline radio/src/targets/common/arm/stm32/aux_serial_driver.cpp:294]: this line should end with aux2SerialRxFifo.size();

**Comment by ajjjjjjjj** [comment]
Thanks @rafolg , done.

**Pull request merged**